src/main/java/uk/gov/hmcts/reform/sscs/util/PdfRequestUtil.java

```diff
@@ -175,31 +176,46 @@ public static DocumentType getPostHearingDocumentType(PostHearingRequestType pos
         }
     }
 
-    public static String getNoticeBody(SscsCaseData caseData, boolean isPostHearingsEnabled, boolean isPostHearingsBEnabled) {
+    public static NoticeIssuedTemplateBody populateNoticeBodySignedByAndSignedRole(SscsCaseData caseData, NoticeIssuedTemplateBody formPayload, boolean isPostHearingsEnabled, boolean isPostHearingsBEnabled) {
+        NoticeIssuedTemplateBody.NoticeIssuedTemplateBodyBuilder formPayloadBuilder = formPayload.toBuilder();
         if (isPostHearingsEnabled) {
             PostHearingReviewType postHearingReviewType = caseData.getPostHearing().getReviewType();
 
             switch (postHearingReviewType) {
                 case SET_ASIDE:
-                    return caseData.getDocumentGeneration().getBodyContent();
+                    formPayloadBuilder.noticeBody(caseData.getDocumentGeneration().getBodyContent());
```

Comment:
caseData.getDocumentGeneration() to seperate var to use across the method

Reply:
Done